api_spec.rb

```diff
@@ -81,35 +81,34 @@ def check_dir_listing_content_type(content_type)
     end
   end
 
-  describe "PUT another JSON object" do
+  describe "updating that JSON object" do
     before do
```

Comment:
I think it would be better to group together the old_listing variables before the put request and the new ones afterwards. Also not all of those variables need to be instance variables (prefixed with `@`) because they're not used outside of the `before` block